virttest/qemu_devices/qcontainer.py

```diff
@@ -2896,6 +2896,20 @@ def throttle_group_define_by_params(self, group_params, name):
         props = json.loads(group_params.get("throttle_group_parameters", "{}"))
         return QThrottleGroup(name, props)
 
+    def thread_context_define_by_params(self, params, name):
+        """
+        Create thread-context object from params.
+        """
+        tc_params = Params()
+        prefix = 'vm_thread_context_'
+        for key in list(params.keys()):
+            if key.startswith(prefix):
+                new_key = key.rsplit(prefix)[1]
+                tc_params[new_key] = params[key]
```

Comment:
Hi @mcasquer 
I did not get the point of this part, why do we still need to filter the parameters by the prefix `vm_thread_context_` to get the related key values?
And how do we handle those params which are without the prefix `vm_thread_context_`? ignore them?
Would you like to provide an example cfg?

IMO, I think that the argument `params` of `thread_context_define_by_params` already represents the related parameters of the `thread-context` object, so we don't filter them again.
like the implementation `throttle_groups`
https://github.com/avocado-framework/avocado-vt/blob/ac53d1ff85f257d6d50fa9411e094b3297eac767/virttest/qemu_vm.py#L2237-L2243
https://github.com/avocado-framework/avocado-vt/blob/ac53d1ff85f257d6d50fa9411e094b3297eac767/virttest/qemu_devices/qcontainer.py#L2903-L2906

Please correct me if I misunderstood, Thanks.

Reply:
@YongxueHong 
If I skip this filtering, then all the params will compose the qemu-kvm cmd line, maybe this is related with the cfg file, what do you think?
e.g.
```bash
[stdlog]     -rf %s,indirect_image_blacklist=/dev/hda[\d]* /dev/sda[\d]* /dev/sg0 /dev/md0,inactivity_treshold=3600,inactivity_watcher=log,vfd_size=1440k,install_virtio=no,enable_libvirtd_debug_log=yes,libvirtd_debug_level=2,libvirtd_log_cleanup=yes,enable_split_libvirtd_feature=no,enable_host_sosreport=no,enable_remote_host_sosreport=no,rpmbuild_path=/root/rpmbuild/,sysprep_required=no,sysprep_options=--operations machine-id,gcov_qemu=no,gcov_qemu_reset=yes,gcov_qemu_collect_cmd_opts=--html,gcov_qemu_compress=no,stress_install_from_repo=no,stress_args=--cpu 4 \
[stdlog]     --io 4 \
[stdlog]     --vm 2 \
[stdlog]     --vm-bytes 256M,download_url_stress=stress/stress-1.0.4.tar.gz,enable_guest_sosreport=no,gluster_managed_by_test=yes,vm_monitor_exit_status=no,kickstart_reboot_bug=no,force_reset_go_down_check=qmp,qemu_stop=on,qemu_preconfig=off,vm_type=qemu,kvm_ver_cmd=uname \
[stdlog]     -r,kvm_userspace_ver_cmd=rpm \
[stdlog]     -qa | grep \
[stdlog]     -E 'qemu-kvm(-(rhev|ma|core))?-[0-9]+\.' | head \
```

Reply:
Hi @mcasquer 
I misunderstand, you are right.